### DISCUSSION TITLE
USB controllers support for ESP32-S2 devices

### DISCUSSION THREAD

**Discussion by anabolyc** [fdivitto/FabGL#224] [Ideas]
Hi Fabrizio 

What do you think about adding support for ESP32-S2 device, it has native USB and would (in theory at least) allow to add USB controllers support, like standard USB keyboards, mice and (perhaps more interesting) USB gaming controllers. This would be especially interesting in the context of PC and gaming emulators.

I'm hardware developer and I'm planning to make development board for ESP32-S2 chip, but I'm lacking skills to provide good software support. Would you be interested in some kind of collaboration in that area?

**Comment by ghost** [comment]
esp32-s2 only has a single core and less RAM.

**Reply by anabolyc** [reply]
I'm aware, but the point is - **it has native USB**, that ESP32 doesn't have. Also there is ESP32-S3 that is dual-core, but for now it is a bit exotic

**Reply by ghost** [reply]
The S3 does not have DACs.
